File: backend/src/main/java/com/travelplanner/api/models/Usuario.java
package com.travelplanner.api.models;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "usuarios")
@Data
@NoArgsConstructor @AllArgsConstructor
@Builder
public class Usuario {

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 100)
    private String nombre;

    @Column(nullable = false, unique = true, length = 150)
    private String email;

    @Column(nullable = false)
    private String password;

    @Column(name = "fecha_registro", nullable = false, updatable = false)
    private LocalDateTime fechaRegistro;

    @PrePersist
    protected void onCreate() {
        this.fechaRegistro = LocalDateTime.now();
    }
}

File: frontend/src/features/usuarios/types/usuario.ts
export interface UsuarioRequestDTO {
  nombre: string;
  email: string;
  password?: string;
}

export interface UsuarioResponseDTO {
  id: number;
  nombre: string;
  email: string;
  fechaRegistro: string;
  roles: string[];
}

export interface LoginRequestDTO {
  email: string;
  password: string;
}

export interface RegistroRequestDTO {
  nombre: string;
  email: string;
  password: string;
}
